lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1373,16 +1376,16 @@ ParseResult FIRStmtParser::parsePostFixIntSubscript(Value &result,
 
   // Make sure the index expression is valid and compute the result type for the
   // expression.
-  auto resultType = result.getType().cast<FIRRTLType>();
-  resultType = SubindexOp::getResultType(resultType, indexNo,
-                                         translateLocation(indexLoc));
+  // TODO: This should ideally be folded into a `tryCreate` method on the
+  // builder (https://llvm.discourse.group/t/3504).
+  std::array<NamedAttribute, 1> attrs = {std::make_pair(
+      builder.getIdentifier("index"), builder.getI32IntegerAttr(indexNo))};
```

Comment:
Scalar values can be passed to arrayref args, so you can just use:

`NamedAttribute attrs = {builder.getIdentifier("index"), builder.getI32IntegerAttr(indexNo)};`

and pass attrs in directly

Reply:
Done! :+1: